File: practise/leetcode/DailyQuestion/202410/1436旅行的终点站.cs
public class Solution {
    public string DestCity(IList<IList<string>> paths) {
        HashSet<string> sa = new HashSet<string>();
        foreach (var p in paths){
            sa.Add(p[0]);
        }
        foreach (var p in paths){
            if(!sa.Contains(p[1])){
                return p[1];
            }
        }
        return "";
    }
}

File: practise/leetcode/DailyQuestion/202410/1436旅行的终点站.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Set<String> sa = new HashSet<>(paths.size());
        Set<String> sb = new HashSet<>();
        for (List<String> p : paths){
            String a = p.get(0);
            String b = p.get(1);
            sb.remove(a);
            if(!sa.contains(b)){
                sb.add(b);
            }
            sa.add(a);
        }
        return sb.iterator().next();
    }
}